master/lustregittagpoller.py

```diff
@@ -0,0 +1,246 @@
+
+    def describe(self):
+        str = ('LustreTagPoller: watching the remote git repository ' +
+               self.repourl + ' for new tags')
+
+        if not self.master:
+            str += " [STOPPED - check log]"
+
+        return str
+
+    # originally from gitpoller
+    def _dovccmd(self, command, args, path=None):
+        def _convert_nonzero_to_fail(res, command, args, path):
+            (stdout, stderr, code) = res
+            if code != 0:
+                raise EnvironmentError('command %s %s in %s on repourl %s failed with exit code %s: %s'
+                                       % (command, args, path, self.repourl, code, stderr))
+
+            return stdout.strip()
+
+        d = utils.getProcessOutputAndValue(self.gitbin, [command] + args, path=path, env=os.environ)
+        d.addCallback(_convert_nonzero_to_fail, command, args, path)
+
+        return d
+
+    def _getRefs(self):
+        """Collect and parse all remote references"""
+
+        d = self._dovccmd('ls-remote', [self.repourl])
+
+        @d.addCallback
+        def parseRefs(rows):
+            refs = {} 
+            for row in rows.splitlines():
+                if '\t' not in row:
+                    continue
+
+                sha, ref = row.split("\t")
+
+                if not self._filter_ref(ref):
+                    refs[ref] = sha
+
+            return refs
+
+        return d
+
+    def _filter_ref(self, ref):
+        """Filter refs that are not tags"""
+        return not ref.startswith("refs/tags/")
+
+    @defer.inlineCallbacks
+    def _process_changes(self, ref, rev):
+        """Process each tag """
+        # initial run, don't parse all history
+        if not self.lastRef:
+            return
+
+        # if a tag has been previously seen, and it's revision hasn't changes, there's nothing to process
+        if ref in self.lastRef and self.lastRef[ref] == rev:
+            return
+
+        log.msg('LustreTagPoller: processing %s(%s) from "%s"'% (ref, rev, self.repourl))
+
+        dl = defer.DeferredList([
+            self._get_commit_timestamp(rev),
+            self._get_commit_author(rev),
+            self._get_commit_files(rev),
+            self._get_commit_comments(rev),
+        ], consumeErrors=True)
+
+        results = yield dl
+
+        # check for failures
+        failures = [r[1] for r in results if not r[0]]
+        if failures:
+            # just fail on the first error; they're probably all related!
+            raise failures[0]
+
+        timestamp, author, files, comments = [r[1] for r in results]
+        yield self.master.addChange(
+            author=author,
+            revision=rev,
+            files=files,
+            comments=comments,
+            when_timestamp=epoch2datetime(timestamp),
+            branch=ref,
+            category=self.category,
+            project=self.project,
+            repository=self.repourl,
+            src='git')
+
+    # originally from gitpoller
```

Comment:
Python is a case-sensitive language.  The class name is GitPoller, not gitpoller.